### ISSUE TITLE
Please wait on itemUpdate

### ISSUE THREAD

**Issue by andyjhall** [tabalinas/jsgrid#68]
Hi,
I am calling updateItem as below however every so often the grid gets stuck with a "please wait" message, i'm assuming that this is because the function is supposed to return the updated item as a JSON object? The page the ajax calls should already return the object however when changing this function to return the ajax call all the fields within the grid are blank, any ideas on why this might be happening?

`updateItem: function(item) {
    updateCostSell();
    $.ajax({
        type: 'POST',
        url: '<url>',
        data: 'ID=' + item.ID + '&desc=' + encodeURIComponent(item.Description) + '&qty=' + encodeURIComponent(item.Quantity) + '&crate=' + encodeURIComponent(item.CostRate) + '&markup=' + encodeURIComponent(item.SellingMarkup) + '&srate=' + encodeURIComponent(item.SellingRate) + '&charge=' + encodeURIComponent(item.Chargable) + '&tsheet=' + " + TimesheetID + @"
    });
},`

returned json:
`{"ID":1497,"EditAll":true,"Description":"Material","Quantity":2,"CostRate":0,"SellingMarkup":10,"SellingRate":0,"Chargable":0,"isMaterial":true,"isMileage":false}`

**Comment by tabalinas** [comment]
Hi, 

If you have async ajax request in the `updateItem` method, you should return a deferred that will be resolved with the resulting data. In your case just write `return $.ajax...`, since $.ajax already returns deferred.

**Comment by andyjhall** [comment]
I've tried doing that however all the columns in the grid row then go blank, I then tried wrapping my ajax call in a console.log but all it seems to return is {readyState: 1}

**Comment by tabalinas** [comment]
Try the following code to see if your request is done properly (check the data field, it looks incorrect):

```javascript
updateItem: function(item) { 
	var d = $.Deferred();
	
	updateCostSell();
	$.ajax({ 
		type: 'POST', 
		url: '<url>', 
		data: 'ID=' + item.ID + '&desc=' + encodeURIComponent(item.Description) + '&qty=' + encodeURIComponent(item.Quantity) + '&crate=' + encodeURIComponent(item.CostRate) + '&markup=' + encodeURIComponent(item.SellingMarkup) + '&srate=' + encodeURIComponent(item.SellingRate) + '&charge=' + encodeURIComponent(item.Chargable) + '&tsheet=' + " + TimesheetID + @" 
	}).done(function(result) {
		console.log("done", result);
		d.resolve(result)
	}).fail(function() {
		console.log("fail");
		d.reject();
	})
	
	return d.promise();
},
```

**Comment by andyjhall** [comment]
Still not sure exactly how promises work but that seems to be working perfectly thanks!
